solutions/hpc/main.tf

```diff
@@ -30,29 +31,194 @@ module "landing_zone" {
+      sudo -E terraform -chdir=${local.remote_ansible_path} init && sudo -E terraform -chdir=${local.remote_ansible_path} apply -auto-approve \
+          -var 'resource_group=${var.resource_group}' \
+          -var 'prefix=${var.prefix}' \
+          -var 'zones=${jsonencode(var.zones)}' \
+          -var 'compute_ssh_keys=${jsonencode(var.compute_ssh_keys)}' \
+          -var 'login_ssh_keys=${jsonencode(var.login_ssh_keys)}' \
+          -var 'storage_ssh_keys=${jsonencode(var.storage_ssh_keys)}' \
+          -var 'vpc=${local.vpc}' \
+          -var 'compute_subnets=${jsonencode(local.compute_subnets)}' \
+          -var 'login_subnets=${jsonencode(local.login_subnets)}' \
+          -var 'storage_subnets=${jsonencode(local.storage_subnets)}' \
+          -var 'protocol_subnets=${jsonencode(local.protocol_subnets)}' \
+          -var 'bastion_security_group_id=${local.bastion_security_group_id}' \
+          -var 'bastion_public_key_content=${local.bastion_public_key_content}' \
+          -var 'bastion_ssh_keys=[]' \
+          -var 'enable_bootstrap=false' \
+          -var 'enable_bastion=false' \
+          -var 'boot_volume_encryption_key=${local.boot_volume_encryption_key}' \
+          -var 'dns_instance_id=${local.dns_instance_id}' \
+          -var 'dns_custom_resolver_id=${local.dns_custom_resolver_id}' \
+          -var 'enable_landing_zone=false' \
+          -var 'ibmcloud_api_key=${var.ibmcloud_api_key}'
+    EOF
+    ]
+  }
+
+  # triggers = {
+  #   always_run = "${timestamp()}"
+  # }
+
+  depends_on = [
+    module.bootstrap,
+    module.bastion,
+    time_sleep.wait_150_seconds
+  ]
+}
+
+resource "null_resource" "bootstrap_resources_destroyer" {
+  count = (var.enable_bastion && var.enable_bootstrap) ? 1 : 0
+  triggers = {
+    bootstrap_private_ip        = local.bootstrap_private_ip
+    bastion_private_key_content = local.bastion_private_key_content
+    bastion_fip                 = local.bastion_fip
+    remote_ansible_path         = local.remote_ansible_path
+    resource_group              = var.resource_group
+    prefix                      = var.prefix
+    zones                       = jsonencode(var.zones)
+    compute_ssh_keys            = jsonencode(var.compute_ssh_keys)
+    login_ssh_keys              = jsonencode(var.login_ssh_keys)
+    storage_ssh_keys            = jsonencode(var.storage_ssh_keys)
+    vpc                         = local.vpc
+    compute_subnets             = jsonencode(local.compute_subnets)
+    login_subnets               = jsonencode(local.login_subnets)
+    storage_subnets             = jsonencode(local.storage_subnets)
+    protocol_subnets            = jsonencode(local.protocol_subnets)
+    bastion_security_group_id   = local.bastion_security_group_id
+    bastion_public_key_content  = local.bastion_public_key_content
+    boot_volume_encryption_key  = local.boot_volume_encryption_key
+    dns_instance_id             = local.dns_instance_id
```

Comment:
manually delete dns instance and try reapply, it should not trigger this. it should create dns only